lib/rules/no-tabs.js

```diff
@@ -21,25 +21,45 @@ module.exports = {
             category: "Stylistic Issues",
             recommended: false
         },
-        schema: []
+        schema: [],
+        fixable: "whitespace"
     },
 
     create(context) {
         return {
             Program(node) {
+                let totalLength = 0;
+
                 context.getSourceLines().forEach((line, index) => {
-                    const match = regex.exec(line);
+                    let match;
+
+                    // Intentional use of conditional assignment - .exec() returns null when it
+                    // cannot find more matches, breaking the loop
+                    // eslint-disable-next-line no-cond-assign
+                    while (match = regex.exec(line)) {
+                        const loc = {
+                            line: index + 1,
+                            column: match.index + 1
+                        };
 
-                    if (match) {
-                        context.report(
+                        context.report({
                             node,
-                            {
-                                line: index + 1,
-                                column: match.index + 1
-                            },
-                            "Unexpected tab character."
-                        );
+                            loc,
+                            message: "Unexpected tab character.",
+
+                            // This function needs to capture the current `match`
+                            // eslint-disable-next-line no-loop-func
```

Comment:
There's a reason this rule shows up in our "eslint:recommended" configuration. I would highly suggest using an IIFE to ensure the correct match value is capture. (You just need to write a test case with multiple replaceable tabs to see the issue with the code as you currently have it.)